### PR TITLE
Feature/dev 1435 python api refactor

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

- Reorganisation of the python api to more closely reflect scala side
- Individual wrappers for all algorithms to make them more discoverable
- Much improved python docs
- JVM no longer starts immediately when importing pyraphtory (only when starting a context or importing the wrappers)
- Methods for tweaking jvm configuration 

### Why are the changes needed?

- General quality of life and doc improvements

### Does this PR introduce any user-facing change? If yes is this documented?

Yes, documentation is updated accordingly

### How was this patch tested?

The usual tests

### CODE REVIEW

File: docs/source/Install/python/pyraphtory.ipynb [hunk 1]
```diff
@@ -206,13 +243,13 @@
        "</div>"
       ],
       "text/plain": [
-       "   timestamp name  outdegree  indegree\n",
-       "0          2    1          2         0\n",
-       "1          2    2          0         1\n",
-       "2          2    3          0         1"
+       "   timestamp  name  outdegree  indegree\n",
```
Comment (Haaroon) [inline docs/source/Install/python/pyraphtory.ipynb:246]: no more names?

**Review by Haaroon — APPROVED** [review]
seems good, i like the new doc changes, much cleaner than before

**Review by rachchan — APPROVED** [review]
looks good to me!

File: docs/source/Install/python/pyraphtory.ipynb (hunk 1 above)
Comment (ljeub-pometry) [reply docs/source/Install/python/pyraphtory.ipynb]: Good spot, fixed

**Pull request merged**